source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -1190,6 +1191,31 @@ message TransformationReplaceLinearAlgebraInstruction {
 
 }
 
+message TransformationReplaceParamsWithStruct {
+
+  // Replaces parameters of the function with a struct containing
+  // values of those parameters.
+
+  // Result ids of parameters to replace.
+  repeated uint32 parameter_id = 1;
+
+  // Fresh id for a new function type. This might be unused if the required type
+  // already exists in the module or if we can change the old type.
+  uint32 fresh_function_type_id = 2;
+
+  // Fresh id for a new struct function parameter to be used as a replacement.
+  uint32 fresh_parameter_id = 3;
+
+  // Fresh id for a struct object containing values of the replaced parameters.
+  // This object will be created before each OpFunctionCall instruction.
+  // Number of elements in this field is equal to the number of callees of the function.
```

Comment:
In fact, instead of having a repeated uint32, can you instead use a repeated Uint32Pair (or - see discussion on chat with Andre - a protobuf map if you think that would be sufficiently deterministic) that maps function call result id to fresh id?

That would mean that we always use the same fresh id for the same function call result, and this makes the transformation more robust.  For example, if during shrinking function calls get removed, or moved around, we will still use the same fresh id for a call site according to its return value id - not its current position in the module.

Reply:
And - just like I mentioned there could be more fresh ids than call sites, it's OK if some of the entries in this map are not used.

Reply:
Fixed in 9a5003e.